dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/EnrollmentPersister.java

```diff
@@ -73,7 +73,7 @@ protected void updateAttributes(
         entityManager,
         preheat,
         enrollment.getAttributes(),
-        preheat.getTrackedEntity(enrollmentToPersist.getTrackedEntity().getUid()),
+        preheat.getTrackedEntity(UID.of(enrollmentToPersist.getTrackedEntity())),
```

Comment:
I am not counting so don't know but it feels like we have lots of changes where I know need todo `UID.of`. This sometimes makes sense if we don't want to deal with errors of invalid `UID`s at this point. If that does not matter we should make it easier to pass around our types and not have to do `UID.of` so often. This is not only related to the `preheat` but any API.

Reply:
I will check this kind of stuff at the end of the changes.
We will need to review the SMS part as well.
One day, hopefully, everything in DHIS2 will be UID and we will not have this problem anymore :grimacing: